pyuvdata/uvdata/mwa_corr_fits.py

```diff
@@ -811,6 +1295,53 @@ def read_mwa_corr_fits(
                         flags,
                     )
 
+            if flag_init:
+                self.flag_init(
+                    num_fine_chans,
+                    edge_width=edge_width,
+                    start_flag=start_flag,
+                    end_flag=end_flag,
+                    flag_dc_offset=flag_dc_offset,
+                )
+
+            # to account for discrepancies between file conventions, in order
+            # to be consistent with the uvw vector direction, all the data must
+            # be conjugated
+            self.data_array = np.conj(self.data_array)
+
+        # wait for LSTs if set in background
+        if proc is not None:
+            proc.join()
+
+        if not self.metadata_only:
+            # reorder polarizations
+            # reorder pols calls check so must come after
+            # lst thread is re-joined.
+            self.reorder_pols()
+
+        # remove bad antennas or flag bad ants
+        # select must be called after lst thread is re-joined
+        if remove_flagged_ants:
+            good_ants = np.delete(self.antenna_numbers, flagged_ants)
+            self.select(antenna_nums=good_ants)
+        else:
+            if not self.metadata_only:
+                # generage baseline flags for flagged ants
+                bad_ant_inds = np.nonzero(
+                    np.logical_or(
+                        np.isin(self.ant_1_array, flagged_ants),
+                        np.isin(self.ant_2_array, flagged_ants),
+                    )
+                )[0]
+                # TODO: Spw axis to be collapsed in future release
+                self.flag_array[bad_ant_inds, :, :, :] = True
+
+        # phasing
+        if phase_to_pointing_center:
+            self.phase(ra_rad, dec_rad)
+
+        if not self.metadata_only:
```

Comment:
The `check` method actually works fine on a metadata_only object, so I don't think you need this if statement.